### DISCUSSION TITLE
How would I upload a folder with subfolders to supabase storage?

### DISCUSSION THREAD

**Discussion by michaelkremenetsky** [supabase/supabase#2486] [Questions] [answered]
Hi, I was trying to upload a nextjs project to supabase storage for a project I am working on but when I drag all the files in it looks like this. 
![image](https://user-images.githubusercontent.com/28833521/126600945-0fea20d5-9bc9-46ea-beab-fe4985beb1da.png)

How would I upload it without it doing this? 
Also is it possible to delete folders from the dashboard because right now when I try to delete the folder this is in it just says "No options"?

**Comment by silentworks** [comment, marked as answer]
Are you trying to host the project in Supabase storage or are you trying to share the project files with Supabase storage?

**Reply by michaelkremenetsky** [reply]
I'm just trying to store them in supabase storage. I'm not trying to use to it to host a site.

**Reply by silentworks** [reply]
I would suggest you zip the files up in that case and upload the zip instead.

**Reply by michaelkremenetsky** [reply]
Ya, that was what I was thinking I just wanted to know if there was a better way than that.

**Reply by inian** [reply]
Preserving folder structure when dragging folders into supabase storage is a pretty good idea! I have filed an issue internally for this. 

We just launched the options to rename and delete folders yesterday.

**Reply by michaelkremenetsky** [reply]
I was also wondering how I upload a project like this using the API?

**Reply by inian** [reply]
We have pushed a fix which should be deployed to the dashboard in a day or two. With the API or client library, this should be straightforward, as long you provide the right path to the [upload call](https://supabase.io/docs/reference/javascript/storage-from-upload), the folder structure will be preserved

**Reply by inian** [reply]
Now launched! https://supabase.io/blog/2021/07/27/storage-beta#directory-uploads
